Update LSP6 standard page, according to `lsp-smart-contracts v0.8.0` (#459)
lukso-network/docs: docs/standards/universal-profile/lsp6-key-manager.md

        <b>value = </b><code>0x0000000000000000000000000000000000000000000000000000000000000004</code>
    </p>

> This permission allows for **editing permissions** of any address that already has some permissions set on the ERC725Account (including itself). One would also need this permission to **change or remove** an existing _controller address_ from `AddressPermission[index]`.

Reviewer: Same recommendation as above. Here are my suggested changes.

```suggestion
This permission allows for **editing permissions** of any address that has some permissions already set on the ERC725Account (including itself). 
The `CHANGEPERMISSIONS` is also needed to:
- 🗑️ **Remove** a controller `address` from the list of `AddressPermissions[]`, meaning:
    - removing its address at a specific index at `AddressPermissions[index]`
    - decreasing the `AddressPermissions[]` Array length (to describe that a controller address has been removed).
- 🖊️ **Edit** an entry in the `AddressPermissions[index]` Array, meaning changing the address stored at a specific index.

> ⚠️ **Warning:** a controller address with `CHANGEPERMISSIONS` can also edit its own permissions. Be cautious when granting this permission!
```

@YamenMerhi @skimaharvey let me know your feedbacks about the **Warning note** for "editing permissions to self". Might be important to document this.

Author: It looks good to me and the mention is important imo

| Permission Type                                        | Description                                                                                                                                                                                         | `bytes32` data key                    |
| ------------------------------------------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------- |
| [**Address Permissions**](#address-permissions)        | defines a set of [**permissions**](#permissions) for an `address`.                                                                                                                                  | `0x4b80742de2bf82acb3630000<address>` |
| [**Allowed Calls**](#allowed-calls)                    | defines a set of interactions (function + address + standard) allowed for a controller address.                                                                                                     | `0x4b80742de2bf393a64c70000<address>` |

> | [**Allowed ERC725Y Data Keys**](#allowed-erc725y-keys) | defines a list of `bytes32` ERC725Y Data Keys an `address` is only allowed to set when doing [`setData(...)`](../smart-contracts/lsp0-erc725-account.md#setdata-array) on the linked ERC725Account. | `0x4b80742de2bf866c29110000<address>` |

Reviewer: Maybe we can remove `bytes32` here, as we have either fixed or dynamic data keys and it is explained further below in its own section?

```suggestion
| [**Allowed ERC725Y Data Keys**](#allowed-erc725y-keys) | defines a list of ERC725Y Data Keys an `address` is only allowed to set via [`setData(...)`](../smart-contracts/lsp0-erc725-account.md#setdata-array) on the linked ERC725Account. | `0x4b80742de2bf866c29110000<address>` |
```

> [See LSP6 for more details](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-6-KeyManager.md#erc725y-data-keys)

The values set under these permission data keys **MUST be of the following format** to ensure correct behavior of these functionalities.

- **Address Permissions**: a `bytes32` value.

> - **Allowed Calls**: an [CompactBytesArray](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-2-ERC725YJSONSchema.md#bytescompactbytesarray) of the tuple `(bytes4,address,bytes4)`.

Reviewer: ```suggestion
- **Allowed Calls**: a [CompactBytesArray](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-2-ERC725YJSONSchema.md#bytescompactbytesarray) of the tuple `(bytes4,address,bytes4)`.
```

> - **Allowed ERC725Y Data Keys**: an [CompactBytesArray](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-2-ERC725YJSONSchema.md#bytescompactbytesarray) of bytes, containing values from `bytes1` to `bytes32`.

Reviewer: ```suggestion
- **Allowed ERC725Y Data Keys**: a [CompactBytesArray](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-2-ERC725YJSONSchema.md#bytescompactbytesarray) of bytes, containing values from `bytes1` to `bytes32`.
```

:::caution

> To **add / remove entries in the list of allowed calls or ERC725Y Data Keys**, the **whole array** should be encoded again and reset. Each update **overrides the entire previous state**.

Reviewer: ```suggestion
To **add / remove entries in the list of allowed calls or ERC725Y Data Keys**, the **whole compact bytes array** should be encoded again and reset. Each update **overrides the entire previous state**.
```

---

### Allowed Calls

You can restrict an address to interact with:

<details>
    <summary>Standards</summary>

These contracts MUST implement the [ERC165](https://eips.ethereum.org/EIPS/eip-165) standard to be able to detect their interfaces.

|   Interface ID   |                     Meaning                      |
| :--------------: | :----------------------------------------------: |
|   `0xffffffff`   |    Interaction with any standard is allowed.     |

> | Other interfaces | Interaction with a specific standard is allowed. |

Reviewer: How about rephrasing the other way around like _"Allowed to interact with any address"_?
Not a big deal, it's still understandable.

</details>

<details>
    <summary>Addresses</summary>

|                   Address                    |                     Meaning                     |
| :------------------------------------------: | :---------------------------------------------: |
| `0xffffffffffffffffffffffffffffffffffffffff` |    Interaction with any address is allowed.     |
|               Other addresses                | Interaction with a specific address is allowed. |

</details>

<details>
    <summary>Functions</summary>

|    Function     |                     Meaning                      |
| :-------------: | :----------------------------------------------: |
|  `0xffffffff`   |    Interaction with any function is allowed.     |

> | Other functions | Interaction with a specific function is allowed. |

Reviewer: agree with @YamenMerhi here

</details>

> To restrict an `<address>` to be allowed to execute any function at this address `0xCA41e4ea94c8fA99889c8EA2c8948768cBaf4bc0` which should

Reviewer: Restrict + allow in the same sentence looks a bit strange imo. Maybe we can break it down in bullet points? Not sure

```suggestion
To restrict an `<address>` to:
- interact only with the LSP0ERC725Account (ERC165 interface ID = ``0x66767497`) deployed at address `0xCA41e4ea94c8fA99889c8EA2c8948768cBaf4bc0`, 
- but allow to call any functions on this contract,

the data key - value pair below can be set in the ERC725Y contract storage.
```

Author: "To allow an `<address>` to execute any function at this address `0xCA41e4ea94c8fA99889c8EA2c8948768cBaf4bc0` which should
be an LSP0 standard `0x66767497`, the data key - value pair below can be set in the ERC725Y contract storage."

Wdyt?

Reviewer: @b00ste 

To allow an `<address>` to execute any function on a LSP0ERC725Account (interface ID `0x66767497`) deployed at address `0xCA41e4ea94c8fA99889c8EA2c8948768cBaf4bc0`, the data key - value pair below can be set in the ERC725Y contract storage.

be an LSP0 standard `0x66767497`, the data key - value pair below can be set in the ERC725Y contract storage.

- **key:** `0x4b80742de2bf393a64c70000<address>`
- **possible values:**

>   - `(bytes4,address,bytes4)[CompactBytesArray]`: an [**CompactBytesArray**](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-2-ERC725YJSONSchema.md#bytescompactbytesarray) of tuple `(bytes4,address,bytes4)` which is created by concatenationg the chosen _function selector_, _address_ and _standard_. E.g. `0x001c66767497CA41e4ea94c8fA99889c8EA2c8948768cBaf4bc0ffffffff`

Reviewer: typo

```suggestion
  - `(bytes4,address,bytes4)[CompactBytesArray]`: an [**CompactBytesArray**](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-2-ERC725YJSONSchema.md#bytescompactbytesarray) of tuple `(bytes4,address,bytes4)` created by concatenating the chosen _function selector_, _address_ and _standard_. E.g. `0x001c66767497CA41e4ea94c8fA99889c8EA2c8948768cBaf4bc0ffffffff`
```

  _CompactBytesArray_: **`0x001cffffffffF70Ce3b58f275A4c28d06C98615760dDe774DE57760d9bba`**
- _Standard_: **any**;  
  _Address_: **`0xd3236aa1B8A4dDe5eA375fd1F2Fb5c354e686c9f`**;  
  _Function_: **any**;  
  _CompactBytesArray_: **`0x001cffffffffd3236aa1B8A4dDe5eA375fd1F2Fb5c354e686c9fffffffff`**

A _CompactBytesArray_ for these 3 interactions would look like this:
`0x`**`001c`**`66767497CA41e4ea94c8fA99889c8EA2c8948768cBaf4bc0ffffffff`**`001c`**`ffffffffF70Ce3b58f275A4c28d06C98615760dDe774DE57760d9bba`**`001c`**`ffffffffd3236aa1B8A4dDe5eA375fd1F2Fb5c354e686c9fffffffff`

</details>

```json
{
  "name": "AddressPermissions:AllowedCalls:<address>",
  "key": "0x4b80742de2bf393a64c70000<address>",
  "keyType": "MappingWithGrouping",
  "valueType": "(bytes4,address,bytes4)[CompactBytesArray]",
  "valueContent": "(Bytes4,Address,Bytes4)"
}
```

![LSP6 Allowed Calls Overview](/img/standards/lsp6/lsp6_allowed_calls_example.jpeg)

:::warning

Allowing a specific standard does not offer security over the inner workings or the correctness of a smart contract. It should be used more as a "mistake prevention" mechanism than a security measure.

:::

:::info

> **If no Allowed Calls are set, a controller cannot interact with any address nor transfer any value (Contract or EOA).**

Reviewer: ```suggestion
**If no Allowed Calls are set (`0x`), a controller cannot interact with any address nor transfer any value (Contract or EOA).**
```

![LSP6 Key Manager Relay Service](/img/standards/lsp6/lsp6-relay-execution.jpeg)

> #### What are you signing?

Reviewer: I suggest we change this title and we make it as subheading 3

```suggestion
### How to sign relay transactions?
```

So that people can find it more easily via the sidebar ⬇️ 
![image](https://user-images.githubusercontent.com/31145285/217234541-c1cae50f-12b3-435f-8390-25d79d9dd4bf.png)

The relay transactions are signed using the [**version 0 of EIP191**](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-191.md#version-0x00). The relay call data that you want to sign **MUST** be the _keccak256 hash digest_ of the following elements _(bytes values)_ concatenated together.

```javascript
0x19 <0x00> <KeyManager address> <LSP6_VERSION> <chainId> <nonce> <value> <payload>
```

| Message elements     | Details                                                                                                                                                                                                                       |
| :------------------- | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `0x19`               | Is a byte intended to ensure that the _relay call signed data_ is not a valid RLP.                                                                                                                                            |

Reviewer: ```suggestion
| `0x19`               | Byte used to ensure that the _relay call signed data_ is not a valid RLP.                                                                                                                                            |
```
